transport/wseb/src/test/java/org/kaazing/gateway/transport/wseb/logging/BasicLoginModuleWithMultiplePrincipalsOnTheSubject.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright 2007-2015, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.transport.wseb.logging;
+
+import java.security.Principal;
+import java.util.Map;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.login.FailedLoginException;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.spi.LoginModule;
+
+import org.kaazing.gateway.security.auth.config.parse.DefaultUserConfig;
+
+
+public class BasicLoginModuleWithMultiplePrincipalsOnTheSubject implements LoginModule {
```

Comment:
Perhaps we can simplify this class by extending BasicLoginModuleWithDefaultUserConfig and overriding the needed methods, i.e. override the commit method in the superclass with the newly added instance of DefaultUserConfig

>
DefaultUserConfig defaultPrincipal2 = new DefaultUserConfig();
defaultPrincipal2.setName("secondPrincipalName");
defaultPrincipal2.setPassword("secondPrincipalPassword");
subject.getPrincipals().add(defaultPrincipal2);